C-Plus-Plus/TowerOfHanoi.cpp

```diff
@@ -0,0 +1,27 @@
+// C++ recursive function to  
+// solve tower of hanoi puzzle  
+#include <bits/stdc++.h> 
+using namespace std; 
+  
+void towerOfHanoi(int n, char from_rod, 
+                    char to_rod, char aux_rod)  
+{  
+    if (n == 1)  
+    {  
+        cout << "Move disk 1 from rod " << from_rod <<  
+                            " to rod " << to_rod<<endl;  
+        return;  
+    }  
+    towerOfHanoi(n - 1, from_rod, aux_rod, to_rod);  
+    cout << "Move disk " << n << " from rod " << from_rod << 
+                                " to rod " << to_rod << endl;  
+    towerOfHanoi(n - 1, aux_rod, to_rod, from_rod);  
+}  
+  
+// Driver code 
+int main()  
+{  
+    int n = 4; // Number of disks  
+    towerOfHanoi(n, 'A', 'C', 'B'); // A, B and C are names of rods  
+    return 0;  
+}  
```

Comment:
Sample Output for n=4 👍
Move disk 1 from rod A to rod B
Move disk 2 from rod A to rod C
Move disk 1 from rod B to rod C
Move disk 3 from rod A to rod B
Move disk 1 from rod C to rod A
Move disk 2 from rod C to rod B
Move disk 1 from rod A to rod B
Move disk 4 from rod A to rod C
Move disk 1 from rod B to rod C
Move disk 2 from rod B to rod A
Move disk 1 from rod C to rod A
Move disk 3 from rod B to rod C
Move disk 1 from rod A to rod B
Move disk 2 from rod A to rod C
Move disk 1 from rod B to rod C

T(n)= O( 2^{(n+1)} - 1), or you can say O(2^n) which is exponentioal
Therefore, the time complexity is O(2^n) or O(a^n) where a is a constant greater than 1. 

 Space complexity is O(n), which is linear.

Reply:
the same thing has been added at the end of cpp file also